aws-android-sdk-s3/src/main/java/com/amazonaws/services/s3/internal/ServiceUtils.java

```diff
@@ -455,7 +455,13 @@ public static boolean skipMd5CheckPerRequest(AmazonWebServiceRequest request) {
             if (om != null && om.getSSEAlgorithm() != null) {
                 return true;
             }
-            return putObjectRequest.getSSECustomerKey() != null;
+            if (putObjectRequest.getSSECustomerKey() != null) {
+                return true;
+            }
+            if (putObjectRequest.getSSEAwsKeyManagementParams() != null &&
+                putObjectRequest.getSSEAwsKeyManagementParams().getAwsKmsKeyId() != null) {
```

Comment:
Include check for getEncryption as well as key is optional.

                ( putObjectRequest.getSSEAwsKeyManagementParams() != null &&
                   ( putObjectRequest.getSSEAwsKeyManagementParams().getAwsKmsKeyId() != null ||  
                     putObjectRequest.getSSEAwsKeyManagementParams.getEncryption() != null )
                 )